### PR TITLE
ci: add github actions and Danger

### PR DESCRIPTION
**Pull request by Chesire**
Added GitHub actions to the project, to build the initial file, and also added Danger to report issues on the PRs.
Danger needs the secret key added still (discussing with @b-reynolds how to get this done).
Some of the CI actions are commented out, so other people can add them to the project, and uncomment as required.

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    #  run: ./gradlew ktlintCheck
+    - name: run lintDebug
+      run: ./gradlew app:lintDebug
+    #- name: run detekt
+    #  run: ./gradlew detekt
+    - name: upload linting results
+      uses: actions/upload-artifact@v2
+      with:
+        name: lint
+        path: build/reports/
```
Comment (Chesire) [inline .github/workflows/android.yml:42]: Actually this might need a change in the `build.gradle` to accommodate this, will update.

**Comment by Chesire** [comment]
Looks like the `gradlew` file needs to be made executable, will update that

**Review by ChrisJanusiewicz — APPROVED** [review]

File: .github/workflows/android.yml (hunk 1 above)
Comment (Chesire) [reply .github/workflows/android.yml:42]: Now done

**Comment by Chesire** [comment]
> Looks like the `gradlew` file needs to be made executable, will update that

Done, now CI will build correctly. Once the Danger API key is added, this should just work.

**Comment by Chesire** [comment]
Ben has added the API key for Danger, so it will now run 👍

**Pull request merged**